crates/sui-core/src/unit_tests/transaction_manager_tests.rs

```diff
@@ -261,33 +299,53 @@ async fn transaction_manager_read_lock() {
         &state.epoch_store_for_testing(),
     );
 
-    // TM should output the 2 read-only transactions eventually.
+    // TM should output the 3 transactions that are only waiting for this object.
     let tx_0 = rx_ready_certificates.recv().await.unwrap().0;
     let tx_1 = rx_ready_certificates.recv().await.unwrap().0;
-    let mut want_digests = vec![transaction_read_0.digest(), transaction_read_1.digest()];
-    want_digests.sort();
-    let mut got_digests = vec![tx_0.digest(), tx_1.digest()];
-    got_digests.sort();
-    assert_eq!(want_digests, got_digests);
+    let tx_2 = rx_ready_certificates.recv().await.unwrap().0;
+    {
+        let mut want_digests = vec![
+            transaction_read_0.digest(),
+            transaction_read_1.digest(),
+            transaction_default.digest(),
+        ];
+        want_digests.sort();
+        let mut got_digests = vec![tx_0.digest(), tx_1.digest(), tx_2.digest()];
+        got_digests.sort();
+        assert_eq!(want_digests, got_digests);
+    }
 
-    // TM should not output default-lock transaction yet.
     sleep(Duration::from_secs(1)).await;
     assert!(rx_ready_certificates.try_recv().is_err());
 
-    assert_eq!(transaction_manager.inflight_queue_len(), 3);
+    assert_eq!(transaction_manager.inflight_queue_len(), 4);
```

Comment:
We can check `inflight_queue_len() == 1` after the notify_commit() calls later. Or maybe we can check `objects_queue_len_and_age()` to make sure there is no txn waiting on shared_object anymore.

Reply:
Done